### ISSUE TITLE
possible to return original and current value from filterDiffByStatus?

### ISSUE THREAD

**Issue by bwgjoseph** [netilon/differify#30]
Hi,

I'm wondering if it is possible to add a flag that allows you to return both the original and current value after calling `filterDiffByStatus` method?

From the `.compare` method, there is both the original and current value, however, the output is more meaningful (in my case), to have 3 block of diff (ADDED, MODIFIED, DELETED). Hence, I do something along this line.

```js
const dif = diff.compare(before, after);
const added = diff.filterDiffByStatus(dif, "ADDED");
const modified = diff.filterDiffByStatus(dif, "MODIFIED");
const deleted = diff.filterDiffByStatus(dif, "DELETED");
```

In a way, whatever is from each `status`, is a subset of `.compare` but in this case, with the filtered result set, the `original` value is no longer available. 

I'm not sure if the result from `filterDiffByStatus` should have `original` value by default, or should it hide behind a flag like 

```js
// 3rd arg to indicate to return the original value which can be false by default
diff.filterDiffByStatus(dif, "ADDED", true);

filterDiffByStatus(result, status, keep = false)
```

However, with that, the type/interface of the result would be different based on the flag. I think typescript should be able to infer correctly and return the correct type though.

What do you think?

---

Side-note: thanks for the quick fix on the other issues that I've reported. Appreciate it, and this library is really great!
Oh yes, I have tested the fixes, and all of them seem to be working great right now.

**Comment by fabian-orue** [comment]
Hi again @bwgjoseph, thank you for your side note! im happy to help others when possible. I have added the feature you requested,  I think that it's a good option to have in mind :). You can check it in the latest released version (3.4.3)

**Comment by bwgjoseph** [comment]
That's really fast! @fabian-orue 

Thank you once again, and I'll report back if I face any issues.
